src/Host/Client/Impl/Host/BrokerClient.cs

```diff
@@ -133,8 +139,8 @@ internal abstract class BrokerClient : IBrokerClient {
                 } catch (UnauthorizedAccessException) {
                     isValidCredentials = false;
                     continue;
-                } catch (HttpRequestException ex) {
-                    throw new RHostDisconnectedException("HTTP error while creating session: " + ex.Message, ex);
+                } catch (BrokerApiErrorException apiex) {
+                    throw new RHostDisconnectedException(apiex.ApiError);
```

Comment:
That's kinda weird, because there are a bunch of places where `RHostDisconnectedException` wraps something else already (e.g. in `RHost.SendAsync`, it wraps `MessageTransportException`), and I'm about to add a few more with session lifetime handling. If we have any place that actually tries to parse its `Message`, it's already broken.